core/stateless/encoding.go

```diff
@@ -70,6 +146,105 @@ func (w *Witness) DecodeRLP(s *rlp.Stream) error {
 	return w.fromExtWitness(&ext)
 }
 
+// EncodeCompressed serializes a witness with optional compression.
+func (w *Witness) EncodeCompressed(wr io.Writer) error {
```

Comment:
Adding metrics for compression time, compression rate, and compression size, could be useful here.

Reply:
[Done](https://github.com/maticnetwork/bor/pull/1648/commits/c27eb8b9c2bce61a40e4ee8ca89cbf3e92f6c153).